### ISSUE TITLE
Don't open using app based on keyword/regex rules

### ISSUE THREAD

**Issue by SodaWithoutSparkles** [1fexd/LinkSheet#313]
### Feature description

Its very annoying when, the app takes over the link (based on the domain) but can't handle it, only to throw it back to LinkSheet again. 

For example, when downloading a release from github mobile, the link (e.g. https://github.com/1fexd/LinkSheet/releases/download/0.0.33/LinkSheet-0.0.33.apk) would be defaulting to github mobile again, but github mobile cant handle it, therefore passing to linksheet again, causing an useless loop.

Another common type of these links is github wiki ([e.g.](https://github.com/BiglySoftware/BiglyBT/wiki)). When you used LinkSheet to handle the link, the Github Android app took the URL but realize that it cant handle the link, then its again a useless loop

**Comment by 1fexd** [comment]
Yeah, I have encountered this behavior myself, but I'm not sure if implementing regex rules are worth the effort.

**Comment by JasonHK** [comment]
Also, a side-effect of this behavior is you'll lose the default app setting once it happens.
